tools/shared.py

```diff
@@ -990,7 +990,9 @@ def __setattr__(self, attr, value):
 
       if attr not in self.attrs:
         msg = "Attempt to set a non-existent setting: '%s'\n" % attr
-        suggestions = ', '.join(difflib.get_close_matches(attr, list(self.attrs.keys())))
+        suggestions = difflib.get_close_matches(attr, list(self.attrs.keys()))
+        suggestions = [suggestion for suggestion in suggestions if suggestion not in self.legacy_settings]
```

Comment:
Nit: I usually prefer single character variable names inside simple list comprehensions to keep them small and (IMHO) more readable.